src/openrct2/world/Location.hpp

```diff
@@ -729,6 +729,16 @@ struct MapRange : public RectRange<CoordsXY>
     }
 };
 
+struct TileRange : public RectRange<TileCoordsXY>
```

Comment:
We are trying to remove more instances of TileCoordsXY not add more. Leave in terms of `CoordsXY` (i.e. just use MapRange)

Reply:
Well I see using MapRange in this scenario as going one of two ways:

1) plugin writers have to be aware of the 32x resolution per tile, i.e. when looking at a tile with the file inspector, the x/y values have to be multiplied by 32
2) OR we begin using MapRange with a completely different basis.

What's the rationale behind trying to remove TileCoordsXY? it seems logical to me to have both classes for the different coordinate systems

Reply:
Mostly to avoid endlessly converting between the two. Entities always use CoordsXY, tile elements use a mixture. (Don't get us started on the numerous places that use X and Y from TileCoordsXYZ, but Z from CoordsXYZ...)

Reply:
so for the plugin interface, should we still send out serialized coordinates in the CoordsXY space (i.e. 32 steps per tile) or in tile space? I think very few plugins would actually make use of the higher precision coordinate system, but perhaps some.

Reply:
Everything at present plugin wise uses CoordsXY as far as I know. There shouldn't be anything that works in terms of TileCoordsXY.